pyadjoint/checkpointing.py

```diff
@@ -164,11 +175,22 @@ def _(self, cp_action, timestep):
             ):
                 for package in self.tape._package_data.values():
                     package.continue_checkpointing()
+            if timestep == 1:
+                # Store the possible global dependencies.
+                for deps in self.tape.timesteps[timestep - 1].checkpointable_state:
+                    self._global_deps.add(deps)
+            else:
+                deps_to_clear = self._global_deps - self._global_deps.intersection(
```

Comment:
I think you might want `set.difference` https://docs.python.org/3/library/stdtypes.html#frozenset.difference